### DISCUSSION TITLE
How to deselect Each object from grouped objects?

### DISCUSSION THREAD

**Discussion by KiDeokKim** [fabricjs/fabric.js#7972] [Q&A] [answered]
Hi everyone, sorry for my bad english writting.
i'm trying to group only the objects that I want out of the objects selected through the drag.
but i need to each objects can be selected. i'm just consider when i select objects from mouse dragged.

one object have some value for separate another object, another object have not this value.
my idea is when event triggered 'selection:created', check object separate value and 
if have this value, discard object in activeobject in array And recalculate grouped object coords.

The reason I'm trying to do this is because specific object don't want rotated. but when i selected group and rotate group, it's rotated.
also specific object is set 'lockRoatation' true value.

any have some idea or function?

**Comment by ShaMan123** [comment]
You can go at it a few ways:
1. `Object#onSelect` - return true to deny it from being selected
2. as you mentioned hook onto an event and remove the objects from the active selection.
3. you can counter transform this object, hooking to the rotating event of the group

**Comment by KiDeokKim** [comment]
Okay, First, I tried to modify array that selected objects in event(selection:created).
but at first when i drag mouse for select objects, all the objects were selected. second time, will separated object is not selected.
However, separated object can't selected as a single object. 

after i checked objects in 'selectable' value, but all object is set true.
what did I do wrong?

Here is my code in hooks "selection:created".
```js
      let selectedObj = fabricCanvas.getActiveObject();
      if (selectedObj._objects !== undefined) {
        let newObjArray = [];
        selectedObj._objects.forEach(obj => {
          if (!obj.isGuideLine) {
            newObjArray.push(obj);
          }
        })
        selectedObj._objects = newObjArray;
        props.setActiveSelection(selectedObj);
      }
      else {
        props.setActiveSelection(selectedObj);
      }
```

**Reply by ShaMan123** [reply]
I don't understand what you wrote.
Code looks okay for fabric<v6

**Reply by KiDeokKim** [reply]
I think my explanation was lacking.
so I'm trying to erase(deselect) some of the selected objects by dragging them with my mouse.

example - i have 3 objects and try to select 3 object in canvas, [A,B,C]
when i select 3 objects, i want discard A and get selected objects [B,C].

but when i tried, selected objects coordinate is weird. (because left and top values of the selected objects are based on the center)
also new selected objects ( [B,C] ) can not selected.

how can I resolve my thoughts?

**Comment by KiDeokKim** [comment, marked as answer]
okay, i soloved it.
i read http://fabricjs.com/manage-selection.
and some code added.
